mass-deployer/internal/parser/parser.go

```diff
@@ -0,0 +1,89 @@
+package parser
+
+import (
+	"fmt"
+	"slices"
+
+	"gopkg.in/yaml.v3"
+)
+
+func ParseConfig(configFile []byte) (Config, error) {
+	conf := Config{}
+
+	err := yaml.Unmarshal(configFile, &conf)
+	if err != nil {
+		return Config{}, err
+	}
+
+	if len(conf.NodeGroups) == 0 {
+		return Config{}, fmt.Errorf("couldn't find any nodes groups to use in deployment")
+	}
+	if len(conf.Vms) == 0 {
+		return Config{}, fmt.Errorf("couldn't find any vms that need to be deployed")
+	}
+	if len(conf.SSHKeys) == 0 {
+		return Config{}, fmt.Errorf("user ssh_keys is invalid, ssh_keys shouldn't be empty")
+	}
+	if len(conf.Mnemonic) == 0 {
```

Comment:
Please validate the mnemonic, also when checking a length of a string we should trim spaces of it `apply for all`
you can use `bip39.IsMnemonicValid(mnemonic)